### PR TITLE
Services with updated logging

### PR DESCRIPTION
**Pull request by jfalkner**
Simple PR branched from #42 so that it is easy to see what services now have the improved logging.

- `GetStatusService` see #42 -- was changed in that PR
- `PbService`
- `DataSetMergerTool`
- `FastaToReference`
- `FastaToReferenceDataSet`
- `ImportDataSet`
- `MovieMetaDataToDataSet` 
- `MovieMetaDataToDataSetRtc` 
- `ReferenceInfoToDataSet` 
- `ValidateDataSet`

Looks like 11 candidates.

```
grep -r --include=*.scala "new OptionParser" . | wc -l
11

$ grep -r --include=*.scala "new OptionParser" .
./smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/DataSetMergerTool.scala:  val parser = new OptionParser[DataSetMergerOptions]("validate-datasets") {
./smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/FastaToReference.scala:  val parser = new OptionParser[FastaToReferenceConfig]("fasta-to-reference") {
./smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/FastaToReferenceDataSet.scala:  val parser = new OptionParser[FastaToReferenceDataSetConfig]("fasta-to-dataset") {
./smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/ImportDataSet.scala:  val parser = new OptionParser[ImportDataSetConfig]("import-dataset") {
./smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/MovieMetaDataToDataSet.scala:  val parser = new OptionParser[MovieMetaDataToDataSetConfig]("movie-metadata-to-dataset") {
./smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/MovieMetaDataToDataSetRtc.scala:  val parser = new OptionParser[MovieMetaDataToDataSetRtcConfig]("movie-metadata-to-dataset-rtc") {
./smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/PbDataSetTools.scala:  lazy val parser = new OptionParser[CustomConfig]("pb-datasets") {
./smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/ReferenceInfoToDataSet.scala:  val parser = new OptionParser[ReferenceConverterConfig]("reference-to-dataset") {
./smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/ValidateDataSet.scala:  val parser = new OptionParser[ValidatorConfig]("validate-datasets") {
./smrt-server-tools/src/main/scala/com/pacbio/secondary/smrttools/tools/GetStatus.scala:  lazy val parser = new OptionParser[GetStatusConfig]("get-status") {
./smrt-server-tools/src/main/scala/com/pacbio/secondary/smrttools/tools/PbService.scala:  lazy val parser = new OptionParser[CustomConfig]("pbservice") {
```

### CODE REVIEW

File: smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/CommandLineUtils.scala [hunk 1]
```diff
@@ -57,10 +59,6 @@ trait timeUtils {
 
 }
 
-trait ToolConfig {
-  val debug: Boolean
-}
-
```
Comment (jfalkner) [inline smrt-analysis/src/main/scala/com/pacbio/secondary/analysis/tools/CommandLineUtils.scala]: Note the delete here. I discovered that the old `ToolConfig` was a subset of the `LoggingConfig` that was added. I updated the code to use `LoggingConfig` and removed a bunch of old repetitive lines that set `debug = false` and did the `--debug` CLI param option parsing.

**Comment by skinner** [comment]
Were you waiting on me on this one?  Circle is showing a compile error, so I assumed you were still working on it.

**Comment by mpkocher** [comment]
@jfalkner you may need to update the tests?

```
[error] (smrt-analysis/test:compileIncremental) Compilation failed
[error] Total time: 298 s, completed May 17, 2016 3:16:56 PM
```

**Comment by jfalkner** [comment]
> Were you waiting on me on this one? Circle is showing a compile error, so I assumed you were still working on it.

A little but not the main reason. Given the other PR merging related to logging, this seems wise to do now.. I was mostly waiting due to needing to remake this branch on latest master. The PR is set to merge to `logging_improvements` and origingally intended to merge before that branch did.

I'll remake this now and see if the test passes.

**Comment by jfalkner** [comment]
#66 is the same code but with the right branch as the merge target. Unfortunately, can't change that in GH post PR creation.

**Pull request closed without merging**